pkg/api/core/v1/client/apps.go

```diff
@@ -138,6 +138,60 @@ func (c *Client) AppShow(namespace string, appName string) (models.App, error) {
 	return resp, nil
 }
 
+// AppGetPart retrieves part of an app (values.yaml, chart, image)
+func (c *Client) AppGetPart(namespace, appName, part, destinationPath string) error {
+
+	endpoint := api.Routes.Path("AppPart", namespace, appName, part)
+	requestBody := ""
+	method := "GET"
+
+	// inlined c.get/c.do to the where the response is handled.
+	uri := fmt.Sprintf("%s%s/%s", c.URL, api.Root, endpoint)
+	c.log.Info(fmt.Sprintf("%s %s", method, uri))
+
+	reqLog := requestLogger(c.log, method, uri, requestBody)
+
+	request, err := http.NewRequest(method, uri, strings.NewReader(requestBody))
+	if err != nil {
+		reqLog.V(1).Error(err, "cannot build request")
+		return err
+	}
+
+	request.SetBasicAuth(c.user, c.password)
+
+	response, err := (&http.Client{}).Do(request)
+
+	if err != nil {
+		reqLog.V(1).Error(err, "request failed")
+		castedErr, ok := err.(*url.Error)
+		if !ok {
+			return errors.New("couldn't cast request Error!")
+		}
+		if castedErr.Timeout() {
+			return errors.New("request cancelled or timed out")
+		}
+
+		return errors.Wrap(err, "making the request")
+	}
+
+	defer response.Body.Close()
+	reqLog.V(1).Info("request finished")
+
+	// Create the file
+	out, err := os.Create(destinationPath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
```

Comment:
Don't you want to catch the potential error using something like 

```
defer func(){
 err = out.Close()
  if err != nil {
    fmt.Println(err)
  } 
}()
```

Reply:
It's sometimes usefull to catch error from defer execution

Reply:
Hm.

Reply:
Not taking it.